### PR TITLE
C++ Inheritance, resolves #295

### LINKED ISSUE #295: [Entry Request] C++: Inheritance
## Description

Write a high-quality "Inheritance" entry in C++. 

C++ topic: https://www.codecademy.com/resources/docs/cpp

## Type of Bug Request

- Typo and/or bug within one or more existing entries.
- Unclear explanation of term(s).
- Out-of-date information (new release, deprecated package, etc).
- Updating the documentation

## Type of Entry Request

- One or more new entries/terms requested.
- Recommend change(s) to documentation (such as existing text or new text)

## Checklist

- [x] The issue is appropriately titled.
- [x] Description is clear and includes supporting text, files, etc.
- [x] A similar issue hasn't already been created.
- [x] No more than 3 entries requested.

## For Maintainers

- [x] Labels are added
- [x] Assignee is added

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by Dusch4593** [comment]
I'll take on the first review 😄

File: content/cpp/concepts/inheritance/inheritance.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  - 'Objects'
+  - 'OOP'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+Inheritance is the ability to base a new class on an existing class, creating a class that starts out with the same properties and methods as the existing class. The class that inherits these properties and methods is called the "derived class" or "sub-class". The class that the sub-class inherits from is called the "base class".
+
+Inheritance is generally used when it's necessary to implement a number of different objects that are all of the same type, sharing several characteristics in common.
```
Comment (Dusch4593) [inline content/cpp/concepts/inheritance/inheritance.md:18]:
```suggestion
Inheritance is the ability to create a new class based on an existing class, starting out with the same existing properties and methods. The class that inherits these properties and methods is called the "derived class" or "sub-class". The class that the sub-class inherits from is called the "base class".

Inheritance is generally used when it's necessary to implement a number of different objects that are all of the same type and share several common characteristics.
```

File: content/cpp/concepts/inheritance/inheritance.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+Title: 'Inheritance'
+Description: 'Inheritance is the ability to base a new class on an existing class, creating a class that starts out with the same properties and methods as the existing class.'
```
Comment (Dusch4593) [inline content/cpp/concepts/inheritance/inheritance.md:3]:
```suggestion
Description: 'Inheritance is the ability to create a new class based on an existing class, starting out with the same existing properties and methods.'
```

File: content/cpp/concepts/inheritance/inheritance.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+Inheritance is the ability to base a new class on an existing class, creating a class that starts out with the same properties and methods as the existing class. The class that inherits these properties and methods is called the "derived class" or "sub-class". The class that the sub-class inherits from is called the "base class".
+
+Inheritance is generally used when it's necessary to implement a number of different objects that are all of the same type, sharing several characteristics in common.
+A good example is a `Shape` class. It might have a `center` property with x,y coordinates, a `weight` property that defines a line width, and a `color` property that would define a fill color. Derived from this base class might be a `Rectangle` which would have `height` and `width` properties, and a `Circle` that would have a `radius` property. Both `Circle` and `Rectangle` would share the properties and methods of their base class, `Shape`, as well as the additional properties and methods that are in their own class definition.
+
+## Syntax
```
Comment (Dusch4593) [inline content/cpp/concepts/inheritance/inheritance.md:21]:
```suggestion

## Example

A good example is a `Shape` class that might include:

- A `center` property with (x, y) coordinates.
- A `weight` property that defines a line width.
- A `color` property that would define a fill color. 

Some possible sub-classes derived from this base class might be:

- A`Rectangle` class which would have `height` and `width` properties.
- A `Circle` class that would have a `radius` property.

Both the `Circle` and `Rectangle` classes would share the properties and methods of their base `Shape` class, `Shape`, in addition to properties and methods unique to their own class definition.
```
Could the sub-classes in this example, perhaps, also have the following:

- One shared method from the base `Shape` class (maybe one for finding the area?)
- At least one unique method for each sub-class (ex. Circumference for `Circle` and Perimeter for `Rectangle`)

File: content/cpp/concepts/inheritance/inheritance.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  public:
+    int radius;
+};
+```
+
+## Access Specifiers
+
+- `public` members of the base class are `public` in the derived class.
+- `protected` members of the base class are accessible by the derived class, but not outside the class.
+- `private` members of the base class aren't accessible to the derived class.
```
Comment (Dusch4593) [inline content/cpp/concepts/inheritance/inheritance.md:50]: I think this should be moved somewhere in the "Syntax" section and changed to a lv 3 header. (`### Access Specifiers`)

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Finished! Take a look and let me know when you finished 😃

**Comment by SSwiniarski** [comment]
@Dusch4593, I've implemented all your suggestions and we're good for review #2.

**Review by Dusch4593 — APPROVED** [review]
LGTM 👍

**Comment by yangc95** [comment]
> LGTM 👍

Ditto 👍

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
Hey @SSwiniarski, this took some time to display but your entry is live here:

https://www.codecademy.com/resources/docs/cpp/inheritance 🎈

ty @Dusch4593 ❗

**Pull request merged**